bridge/src/lib.rs

```diff
@@ -168,8 +183,10 @@ methods!(
         let bytes = unwrap_bytes(proto.map_err(VM::raise_ex).unwrap());
         let ruby_callback = VM::block_proc();
         let callback = move |result: WorkerResult| {
-            result.map_err(|e| raise_bridge_exception(&e.to_string())).unwrap();
-            ruby_callback.call(&[NilClass::new().to_any_object()]);
+            match result {
+                Ok(_) => ruby_callback.call(&[ruby_nil()]),
+                Err(e) => ruby_callback.call(&[ruby_nil(), wrap_worker_error(&e).to_any_object()])
+            };
```

Comment:
Rather than dupe these lines twice, you could implement:

`impl From<WorkerResult> for &[xxx]`

Where `xxx` is the type inside these slices, some ruby object thing. Then both of these lines can become
`ruby_callback.call(result.into())`

Reply:
Ah, interesting, let me look into this

Reply:
Tracking here — #95